packages/pwa-kit-runtime/src/ssr/server/build-remote-server.js

```diff
@@ -473,6 +486,194 @@ export const RemoteServerFactory = {
         })
     },
 
+    /**
+     * @private
+     */
+    _setupBasePathMiddleware(app) {
+        // Cache the express route regexes to avoid re-calculating them
+        // on every request.
+        let expressRouteRegexes
+
+        const removeBasePathFromPath = (path) => {
+            const regex = new RegExp(`^${getEnvBasePath()}(/|$)`)
+            return path.replace(regex, '/')
+        }
+
+        /*
+         * '?' have different meanings in express and path-to-regexp so we need
+         * to convert the express route to something path-to-regexp can process
+         *
+         * Express uses ? to mark the preceeding character as optional
+         * For example: the express route /abc/def? corresponds to
+         * the regex /abc\/de(f)?/
+         *
+         * However, Path-to-regexp uses ? to mark a parameter as optional
+         * For example: /abc/:def? means the :def parameter is optional
+         *
+         * In path-to-regexp, a route with ? that is not preceeded by
+         * a : must wrap the optional character in a paranthesis to
+         * be condidered valid
+         *
+         * In other words, path-to-regexp allows /abc/de(f)? but
+         * /abc/def? is considered invalid.
+         */
+        const convertExpressOptionalChars = (routePattern) => {
```

Comment:
As mentioned in the code comments, express v4 and path-to-regex handle `?` differently. So we have to manually massage the incoming express route into something that path-to-regex can handle.